packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -831,8 +832,16 @@ abstract class FlutterCommand extends Command<void> {
             'This flag is only supported on "--release" builds. When building for Android, a single '
             'ABI must be specified at a time with the "--target-platform" flag. When building for iOS, '
             'only the symbols from the arm64 architecture are used to analyze code size.\n'
+            'By default, the intermediate output files will be placed in a temporary directory in the '
+            'build directory. This can be overriden with the --${FlutterOptions.kCodeSizeDirectory} option.\n'
             'This flag cannot be combined with "--${FlutterOptions.kSplitDebugInfoOption}".'
     );
+
+    argParser.addOption(
+      FlutterOptions.kCodeSizeDirectory,
+      help: 'The location to write code size analysis files. If this is not specified, files '
+            'are written to a temporary directory under the build directory.'
```

Comment:
nit: not a temporary directory, since that would be in a tmp folder, its more of a unqiue/arbitrary directory

Reply:
changing to `transient`.